ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java

```diff
@@ -1085,28 +1102,77 @@ public void waitForInitNotification() throws InterruptedException {
       // wait slightly more than 5 mins to ensure routerUpdater refreshes the snapshot.
       if (!routingTableInitLatch.await(320, TimeUnit.SECONDS)) {
         throw new IllegalStateException(
-            "Initial routing table change from " + dcOrClusterName + " didn't come within 5 mins");
+            "Initial routing table change from helix cluster " + helixClusterName + "in dc " + dcName
+                + " didn't come within 5 mins");
       }
     }
 
     public void registerClusterMapListener(ClusterMapChangeListener clusterMapChangeListener) {
       clusterMapChangeListeners.add(clusterMapChangeListener);
     }
 
+    /**
+     * Update partition to resource mapping from Ideal states.
+     * @param idealStates list of ideal states for various resources.
+     * @param dcName data center for which this mapping corresponds to. If it is null, the mapping is for resources in
+     */
+    private void updatePartitionResourceMappingFromIdealStates(Collection<IdealState> idealStates, String dcName) {
+      // rebuild the entire partition-to-resource map in current dc
+      for (IdealState state : idealStates) {
+        String resourceName = state.getResourceName();
+        for (String partition : state.getPartitionSet()) {
+          if (dcName != null) {
+            ConcurrentHashMap<String, String> partitionToResourceName =
+                partitionToResourceNameByDc.computeIfAbsent(dcName, k -> new ConcurrentHashMap<>());
+            partitionToResourceName.put(partition, resourceName);
+          } else {
+            List<String> resourceNames =
+                globalPartitionToResourceNames.computeIfAbsent(partition, k -> new CopyOnWriteArrayList<>());
+            resourceNames.add(resourceName);
+          }
```

Comment:
we are not listening on the IdealState change in the Aggregated view, we don't need this logic.

Reply:
Done.